app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/ReadFileTask.java

```diff
@@ -78,38 +82,24 @@ protected ReturnedValues doInBackground(Void... params) {
           if (fileAbstraction.uri == null)
             throw new NullPointerException("Something went really wrong!");
 
-          inputStream = contentResolver.openInputStream(fileAbstraction.uri);
+          if (fileAbstraction.uri.getAuthority().equals(AppConfig.getInstance().getPackageName())) {
+            DocumentFile documentFile =
+                DocumentFile.fromSingleUri(AppConfig.getInstance(), fileAbstraction.uri);
+            if (documentFile != null && documentFile.exists() && documentFile.canRead())
```

Comment:
Better check for `documentFile.canWrite()`  instead of `documentFile.canRead()` so that if we can't write to file, we'll fallback to creating cache file.